### ISSUE TITLE
Variable scope issue with Mixins

### ISSUE THREAD

**Issue by clifff** [sass/libsass#1799]
I'm pretty lost on how to describe what exactly is going on here, but I noticed differences in my applications SCSS when using Ruby sass vs libass, and have simplified it down to this:

### Test

```
$list: 'foo', 'bar';

@mixin iterator($x){
  @each $el in $list {
    @content
  }
}

@each $outside_el in $list {
  $selector: ();
  @include iterator($outside_el){
    @each $y in $list {
      $selector: append($selector, unquote('.l-#{$y}'), 'comma');
    }
  }
  #{$selector}{
    &.tux{
      a: b;
    }
  }
}
```

### Output from Ruby Sass
```
.l-foo.tux, .l-bar.tux, .l-foo.tux, .l-bar.tux {
  a: b; }

.l-foo.tux, .l-bar.tux, .l-foo.tux, .l-bar.tux {
  a: b; }
```

### Output from libass
```
.tux {
  a: b; }
```

I've noticed that if the mixin is replaced with calling that code directly, the result is as expected though.

**Comment by saper** [comment]
I wonder if this is another instance of #1757

**Comment by xzyfer** [comment]
This is fixed on master and will be in 3.3.3.

**Comment by xzyfer** [comment]
Fixed by https://github.com/sass/libsass/pull/1748. The incorrect output was due to variable scope not being respected in some case causing variable to be overwritten.
